src/api.js

```diff
@@ -2001,11 +2000,12 @@ api.declare({
     'page. You may limit this with the query-string parameter `limit`.',
   ].join('\n'),
 }, async function(req, res) {
-  let continuation = req.query.continuationToken || null;
-  let limit = parseInt(req.query.limit || 1000, 10);
+  const continuation = req.query.continuationToken || null;
+  const isLimitValid = req.query.limit && req.query.limit < 1000;
+  const limit = parseInt(isLimitValid ? req.query.limit : 1000, 10);
```

Comment:
Just do:
`const limit = Math.min(1000, parseInt(req.query.limit || 1000, 10));`

Notice that: `req.query.limit < 1000` is actually string and int comparison...It works, but let's avoid it :)